x-pack/plugins/security/server/lib/authentication/providers/token.js

```diff
@@ -356,7 +362,30 @@ export class TokenAuthenticationProvider {
       // it's called with this request once again down the line (e.g. in the next authentication provider).
       delete request.headers.authorization;
 
+      // If refresh fails with `400` then refresh token is no longer valid and we should clear session
+      // and redirect user to the login page to re-authenticate.
+      if (getErrorStatusCode(err) === 400 && canRedirectRequest(request)) {
```

Comment:
Save question as above...

Is it possible that when we do the access token logic, we get back the 401 because the token document exists and it's expired, and then before we get here the refresh token is deleted and we get back a 500 with the message body denoting the token being deleted?

Reply:
See reply above, tl;dr it shouldn't be possible.

Reply:
Thanks for looking into this!